src/main/java/com/synopsys/integration/alert/channel/slack/parser/SlackChannelEventParser.java

```diff
@@ -0,0 +1,190 @@
+        this.slackChannelMessageParser = slackChannelMessageParser;
+        this.restChannelUtility = restChannelUtility;
+    }
+
+    public List<Request> createRequests(DistributionEvent event) throws IntegrationException {
+        FieldAccessor fields = event.getFieldAccessor();
+
+        String webhook = fields.getString(SlackDescriptor.KEY_WEBHOOK).orElse("");
+        String channelName = fields.getString(SlackDescriptor.KEY_CHANNEL_NAME).orElse("");
+
+        if (StringUtils.isBlank(webhook) || StringUtils.isBlank(channelName)) {
+            Map<String, String> fieldErrors = new HashMap<>();
+            if (StringUtils.isBlank(webhook)) {
+                fieldErrors.put(SlackDescriptor.KEY_WEBHOOK, "Missing Webhook URL");
+            }
+            if (StringUtils.isBlank(channelName)) {
+                fieldErrors.put(SlackDescriptor.KEY_CHANNEL_NAME, "Missing channel name");
+            }
+            throw new AlertFieldException(fieldErrors);
+        } else {
+            Optional<String> channelUsername = fields.getString(SlackDescriptor.KEY_CHANNEL_USERNAME);
+
+            MessageContentGroup eventContent = event.getContent();
+            if (eventContent.isEmpty()) {
+                return List.of();
+            } else {
+                Map<String, String> requestHeaders = new HashMap<>();
+                requestHeaders.put("Content-Type", "application/json");
+
+                String actualChannelUsername = channelUsername.orElse(SLACK_DEFAULT_USERNAME);
+                List<String> mrkdwnMessagePieces = slackChannelMessageParser.createMessagePieces(eventContent);
+                return createRequestsForMessage(channelName, actualChannelUsername, webhook, mrkdwnMessagePieces, requestHeaders);
+            }
+        }
+    }
+
+    private List<Request> createRequestsForMessage(String channelName, String channelUsername, String webhook, List<String> mrkdwnMessagePieces, Map<String, String> requestHeaders) {
+        List<String> mrkdwnMessageChunks = splitMessages(mrkdwnMessagePieces);
+        return mrkdwnMessageChunks
+                   .stream()
+                   .filter(StringUtils::isNotBlank)
+                   .map(message -> getJsonString(message, channelName, channelUsername))
+                   .map(jsonMessage -> restChannelUtility.createPostMessageRequest(webhook, requestHeaders, jsonMessage))
+                   .collect(Collectors.toList());
+    }
+
+    private String getJsonString(String htmlMessage, String channel, String username) {
+        JsonObject json = new JsonObject();
+        json.addProperty("text", htmlMessage);
+        json.addProperty("channel", channel);
+        json.addProperty("username", username);
+        json.addProperty("mrkdwn", true);
+
+        return json.toString();
+    }
+
+    private List<String> splitMessages(List<String> messagePieces) {
```

Comment:
Might be beyond the scope of this PR but MessageSplitter can be used in place of the splitMessages since MessageSplitter was based on this code written for Slack.

Reply:
I'll create a ticket for this this.